HIP/hip-0000-event-serialization.md

```diff
@@ -0,0 +1,227 @@
+have not changed may be omitted.
+
+### NodeId
+
+```
+
++------------------------------------------------------------------------+
+| NodeId Object                                                          |
++------------+---------+---------------------------------+---------------+
+| # of bytes | Type    | Description                     | Typical Value |
++------------+---------+---------------------------------+---------------+
+| 4 bytes    | Integer | Serialization version of object | 0x00000001    |
++------------+---------+---------------------------------+---------------+
+| 8 bytes    | Long    | Node id value                   |               |
++------------+---------+---------------------------------+---------------+
+```
+
+### Hash Object
+
+```
++------------------------------------------------------------------------+
+| Hash Object                                                            |
++------------+---------+---------------------------------+---------------+
+| # of bytes | Type    | Description                     | Typical Value |
++------------+---------+---------------------------------+---------------+
+| 4 bytes    | Integer | Serializaiton version of object | 0x00000001    |
++------------+---------+---------------------------------+---------------+
+| 4 bytes    | Integer | Hash type                       | 0x58ff811b    |
++------------+---------+---------------------------------+---------------+
+| 4 bytes    | Integer | Length of hash                  | 0x00000030    |
++------------+---------+---------------------------------+---------------+
+| N bytes    | byte[]  | Hash value (N = length of hash) |               |
++------------+---------+---------------------------------+---------------+
+```
+
+### EventDescriptor
+
+```
++------------------------------------------------------------------------+
+| EventDescriptor Object                                                 |
++------------+---------+---------------------------------+---------------+
+| # of bytes | Type    | Description                     | Typical Value |
++------------+---------+---------------------------------+---------------+
+| 4 bytes    | Integer | Serialization version of object | 0x00000003    |
++------------+---------+---------------------------------+---------------+
+| H bytes    | Hash    | The hash of the event           |               |
++------------+---------+---------------------------------+---------------+
+| 12 bytes   | NodeId  | The node that created the event |               |
++------------+---------+---------------------------------+---------------+
+| 8 bytes    | Long    | The generation of the event     |               |
++------------+---------+---------------------------------+---------------+
+| 8 bytes    | Long    | The roster round of the event   |               |
++------------+---------+---------------------------------+---------------+
+```
+
+### BaseEventHashedData
```

Comment:
Are name changes up for discussion?

- `BaseEventHashedData` could be renamed to just `EventHashedData`. Not sure what "base" has to do with it
- Similarly, `BaseEvent` could just be `Event`. After all, we do just call it an "event" everywhere, not a "base event"

Reply:
I'm keen on the name change.